vendor/src/github.com/docker/go-connections/nat/nat.go

```diff
@@ -146,9 +146,31 @@ func ParsePortSpecs(ports []string) (map[Port]struct{}, map[Port][]PortBinding,
 			rawPort = fmt.Sprintf(":%s", rawPort)
 		}
 
-		parts, err := PartParser(portSpecTemplate, rawPort)
-		if err != nil {
-			return nil, nil, err
+		parts := make(map[string]string)
+		if rawPort != "" && rawPort[0] == '[' {
```

Comment:
> I don’t see what using net.SplitHostPort() would buy us in this case. I don’t think it makes the code any shorter or handles any more cases than what we handle currently.

I didn't delve into it deeply; but thought to mention it, in case that code was better tested and/or handled corner case that we didn't think of.